toast-tk-interpret/src/main/java/io/toast/tk/action/interpret/web/InterpretationProvider.java

```diff
@@ -9,35 +9,43 @@
 
 import io.toast.tk.swing.agent.interpret.MongoRepositoryCacheWrapper;
 
-
 public class InterpretationProvider {
-	
+
 	private MongoRepositoryCacheWrapper mongoRepoManager;
 
-	private Map<String, IActionInterpret> map;
-	
+
 	@Inject
 	public InterpretationProvider(MongoRepositoryCacheWrapper mongoRepoManager){
 		this.mongoRepoManager = mongoRepoManager;
-		map = new HashMap<String, IActionInterpret>();
-		map.put("a", new WebClickInterpret(this.mongoRepoManager));
-		map.put("select", new SelectInterpret(this.mongoRepoManager));
-		map.put("button", new WebClickInterpret(this.mongoRepoManager));
-		map.put("text", new KeypressInterpret(this.mongoRepoManager));
-		map.put("open", new OpenNewPageInterpret(this.mongoRepoManager));
+
 	}
 
-	public IActionInterpret getSentenceBuilder(
-		String type) {
+	public IActionInterpret getSentenceBuilder(String type) {
 		return getInterpretFor(type);
 	}
 
-	private IActionInterpret getInterpretFor(
-		final String type) {
+	private IActionInterpret getInterpretFor(final String type) {
 		String formatedType = type;
-		if(formatedType != null && formatedType.contains(":")){
+		if (formatedType != null && formatedType.contains(":")) {
 			formatedType = StringUtils.split(type, ":")[0];
 		}
-		return map.get(formatedType);
+		return get(formatedType);
+	}
+
+	private IActionInterpret get(String type) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "get" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.talanlabs.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)